snakemake/executors.py

```diff
@@ -2168,3 +2171,878 @@ def run_wrapper(
+           NVIDIA P100, and the same T4 for a graphical workstation. Since
+           this isn't a graphical workstation use case, we choose the 
+           accelerator that has >= to the maximumCardsPerInstace
+        """
+        if not gpu_count or gpu_count == 0:
+            return
+
+        accelerators = self._retry_request(
+            self._compute_cli.acceleratorTypes().list(project=self.project, zone=zone)
+        )
+
+        # Filter down to those with greater than or equal to needed gpus
+        keepers = {}
+        for accelerator in accelerators.get("items", []):
+
+            # We don't need a virtual workstation
+            if accelerator["name"].endswith("vws"):
+                continue
+
+            if gpu_model and accelerator["name"] != gpu_model:
+                continue
+
+            if accelerator["maximumCardsPerInstance"] >= gpu_count:
+                keepers[accelerator["name"]] = accelerator
+
+        # If no matches available, exit early
+        if not keepers:
+            if gpu_model:
+                raise WorkflowError(
+                    "An accelerator in zone {zone} with model {model} cannot "
+                    " be satisfied, so there are no options "
+                    "available.".format(zone=zone, model=gpu_model)
+                )
+            else:
+                raise WorkflowError(
+                    "An accelerator in zone {zone} cannot be satisifed, so "
+                    "there are no options available.".format(zone=zone)
+                )
+
+        # Find smallest (in future the user might have preference for the type)
+        smallest = list(keepers.keys())[0]
+        max_gpu = keepers[smallest]["maximumCardsPerInstance"]
+
+        # This should usually return P-100, which would be preference (cheapest)
+        for name, accelerator in keepers.items():
+            if accelerator["maximumCardsPerInstance"] < max_gpu:
+                smallest = name
+                max_gpu = accelerator["maximumCardsPerInstance"]
+
+        return keepers[smallest]
+
+    def _set_snakefile(self):
+        """The snakefile must be a relative path, which cannot be reliably
+           derived from the self.workflow.snakefile as we might have moved
+           execution into a temporary directory, and the initial Snakefile
+           was somewhere else on the system.
+        """
+        from snakemake import SNAKEFILE_CHOICES
+
+        for snakefile in SNAKEFILE_CHOICES:
+            if os.path.exists(os.path.join(self.workdir, snakefile)):
+                self.snakefile = snakefile
+                break
+
+    def _set_workflow_sources(self, warning_size_gb=20):
+        """We only add files from the working directory that are config related
+           (e.g., the Snakefile or a config.yml equivalent), or checked into git. 
+           Additionally, given that we encourage these packages to be small,
+           we set a warning at 20GB.
```

Comment:
Individual files in the sources should be much less. Set the limit to 200MB.

Reply:
Please move this into a `Workflow.check_source_sizes` function, and just call it from here. This is helpful as well for the kubernetes and the tibanna executor.

Reply:
All set.